spring-data-relational/src/test/java/org/springframework/data/relational/core/mapping/DefaultAggregatePathUnitTests.java

```diff
@@ -452,43 +453,89 @@ void getRequiredPersistentPropertyPath() {
 		});
 	}
 
-	@Test // GH-1525
-	void getEffectiveIdColumnName() {
+	@Test
+	// GH-1525
+	void getLength() {
 
 		assertSoftly(softly -> {
+			softly.assertThat(path().getLength()).isEqualTo(1);
+			softly.assertThat(path().stream().collect(Collectors.toList())).hasSize(1);
 
-			softly.assertThat(path().getTableInfo().effectiveIdColumnName()).isEqualTo(quoted("ENTITY_ID"));
-			softly.assertThat(path("second.third2").getTableInfo().effectiveIdColumnName()).isEqualTo(quoted("DUMMY_ENTITY"));
-			softly.assertThat(path("withId.second.third").getTableInfo().effectiveIdColumnName())
-					.isEqualTo(quoted("WITH_ID"));
-			softly.assertThat(path("withId.second.third2.value").getTableInfo().effectiveIdColumnName())
-					.isEqualTo(quoted("WITH_ID"));
+			softly.assertThat(path("second.third2").getLength()).isEqualTo(3);
+			softly.assertThat(path("second.third2").stream().collect(Collectors.toList())).hasSize(3);
+
+			softly.assertThat(path("withId.second.third").getLength()).isEqualTo(4);
+			softly.assertThat(path("withId.second.third2.value").getLength()).isEqualTo(5);
 		});
 	}
 
-	@Test // GH-1525
-	void getLength() {
+	@Test // GH-574
+	void getTail() {
 
-		assertThat(path().getLength()).isEqualTo(1);
-		assertThat(path().stream().collect(Collectors.toList())).hasSize(1);
+		assertSoftly(softly -> {
 
-		assertThat(path("second.third2").getLength()).isEqualTo(3);
-		assertThat(path("second.third2").stream().collect(Collectors.toList())).hasSize(3);
+			softly.assertThat((Object) path().getTail()).isEqualTo(null);
+			softly.assertThat((Object) path("second").getTail()).isEqualTo(null);
+			softly.assertThat(path("second.third").getTail().toDotPath()).isEqualTo("third");
+			softly.assertThat(path("second.third.value").getTail().toDotPath()).isEqualTo("third.value");
+		});
+	}
+
+	@Test // GH-74
+	void append() {
+
+		assertSoftly(softly -> {
+
+			softly.assertThat(path("second").append(path()).toDotPath()).isEqualTo("second");
```

Comment:
There's so much to read in these test style that the actual test fixture and test code becomes invisible.

Please use much simpler assertions, otherwise I don't understand what is happening here.

Applies also for many other places in this project.

Reply:
I tried to improve the readability of the tests with a custom assertion.